functions/index.js

```diff
@@ -0,0 +1,99 @@
+//
+// This source file is part of the StudyApplication based on the Stanford Spezi Template Application project
+//
+// SPDX-FileCopyrightText: 2023 Stanford University
+//
+// SPDX-License-Identifier: MIT
+//
+
+const {onCall} = require("firebase-functions/v2/https");
+const {logger, https} = require("firebase-functions/v2");
+const {FieldValue} = require("firebase-admin/firestore");
+const admin = require("firebase-admin");
+const {beforeUserCreated} = require("firebase-functions/v2/identity");
+
+admin.initializeApp();
+
+exports.checkInvitationCode = onCall(async (request) => {
+  if (!request.auth || !request.auth.uid) {
+    throw new https.HttpsError(
+        "unauthenticated",
+        "User is not properly authenticated.",
+    );
+  }
+
+  const {invitationCode} = request.data;
+  const userId = request.auth.uid;
+  const firestore = admin.firestore();
+
+  logger.debug(`User (${userId}) -> PAWS, InvitationCode ${invitationCode}`);
+
+  try {
+    // Based on https://github.com/StanfordSpezi/SpeziStudyApplication/blob/main/functions/index.js
+    const invitationCodeRef = firestore.doc(`invitationCodes/${invitationCode}`);
+    const invitationCodeDoc = await invitationCodeRef.get();
+
+    if (!invitationCodeDoc.exists || invitationCodeDoc.data().used) {
```

Comment:
What do we think about checking whether the used invitation code belongs to the user who triggered this request, @PSchmiedmayer? Like so:
```js
if (!invitationCodeDoc.exists || (invitationCodeDoc.data().used && invitationCodeDoc.data().usedBy == userId)) { ... }
```

Reply:
Sure, sounds like a fair logic. In the end this case should never happen in a production use case as the sign up method should only be called a single time but this could be useful for testing.